### ISSUE TITLE
please help with install issue

### ISSUE THREAD

**Issue by gdfarr** [harvard-library/aspace-import-excel#46]
Hi Bobbi,

I attempted to install your aspace-import-excel plugin on a windows platform, using version 2.1.1 ... I am operating with 2.5 version of ASpace. Unfortunately, after what I thought was a success in running the initialize-plugin.bat script and getting confirmation that 4 gems were installed, there still seems to be an obstacle in getting ASpace back online. The server message I received was:

Bundler could not find compatible versions for gem "bundler": In Gemfile: rails (= 5.0.1) java was resolved to 5.0.1, which depends on bundler (< 2.0, >= 1.3.0) java Current Bundler version: bundler (2.0.1) This Gemfile requires a different version of Bundler. Perhaps you need to update Bundler by running `gem install bundler`? Could not find gem 'bundler (< 2.0, >= 1.3.0)', which is required by gem 'rails (= 5.0.1)', in any of the sources. from C:/archivesspace/plugins/aspace-import-excel/gems/gems/bundler-2.0.1/lib/bundler/resolver.rb:56:in `start' from C:/archivesspace/plugins/aspace-import-excel/gems/gems/bundler-2.0.1/lib/bundler/resolver.rb:22:in `resolve' from C:/archivesspace/plugins/aspace-import-excel/gems/gems/bundler-2.0.1/lib/bundler/definition.rb:258:in `resolve' from C:/archivesspace/plugins/aspace-import-excel/gems/gems/bundler-2.0.1/lib/bundler/definition.rb:170:in `specs' from C:/archivesspace/plugins/aspace-import-excel/gems/gems/bundler-2.0.1/lib/bundler/definition.rb:237:in `specs_for' from C:/archivesspace/plugins/aspace-import-excel/gems/gems/bundler-2.0.1/lib/bundler/definition.rb:226:in `requested_specs' from C:/archivesspace/plugins/aspace-import-excel/gems/gems/bundler-2.0.1/lib/bundler/runtime.rb:108:in `block in requested_specs' from C:/archivesspace/plugins/aspace-import-excel/gems/gems/bundler-2.0.1/lib/bundler/runtime.rb:20:in `setup' from C:/archivesspace/plugins/aspace-import-excel/gems/gems/bundler-2.0.1/lib/bundler.rb:107:in `setup' from C:/archivesspace/plugins/aspace-import-excel/gems/gems/bundler-2.0.1/lib/bundler/setup.rb:20:in `<main>' from org/jruby/RubyKernel.java:961:in `require' from uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:1:in `(root)' from uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:133:in `<main>' from org/jruby/RubyKernel.java:961:in `require' from C:/archivesspace/data/tmp/jetty-0.0.0.0-8080-frontend.war-_-any-/webapp/WEB-INF/config/boot.rb:10:in `(root)' from uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:1:in `(root)' from uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:55:in `require' from uri:classloader:/jruby/rack/rails/environment3.rb:23:in `<main>' from C:\archivesspace\launcher\launcher.rb:92:in `start_server' from C:\archivesspace\launcher\launcher.rb:168:in `main' from C:\archivesspace\launcher\launcher.rb:261:in `<main>'

You're seeing this error because you use JRuby::Rack::ErrorApp::ShowStatus

Could you possibly help me get this issue resolved?

Many thanks in advance,
Greg

Gregory D. Farr

Archivist

Episcopal Church in Connecticut

The Commons

290 Pratt Street | Box 52 | Meriden CT 06450
p 203-639-3501 x135

f 203-235-1008
gfarr@episcopalct.org

**Comment by bobbi-SMR** [comment]
Hi, Greg,

I'm a little hampered by the fact that I don't have a Windows version of Aspace to experiment with.

We're looking into the possibility that the fact that the Bundler team has released a new version of Bundler (https://bundler.io/blog/2019/01/03/announcing-bundler-2.html) may be causing a problem, and we're trying to figure out a way to resolve this. 

Stay tuned!  I'm sorry about the delay.

**Comment by pobocks** [comment]
Hi Greg,

Would you try installing this version of the plugin: https://github.com/harvard-library/aspace-import-excel/archive/issue_46.zip

And letting me know if you have the same problem?

**Comment by gdfarr** [comment]
Thank you! Am on the road today but will try it out later today. Much appreciated!
Best regards,
Greg

Get Outlook for iOS<https://aka.ms/o0ukef>

________________________________

**Comment by gdfarr** [comment]
Good morning,

Thanks again for responding to my posted issue. I did try installing the plugin version that you provided. I seem to be successfully initializing the plugin, but apparently the gem bundler package that I am receiving is not compatible?

Here again is the browser error message (500) that returns:

Bundler could not find compatible versions for gem "bundler": In Gemfile: rails (= 5.0.1) java was resolved to 5.0.1, which depends on bundler (< 2.0, >= 1.3.0) java Current Bundler version: bundler (2.0.1) This Gemfile requires a different version of Bundler. Perhaps you need to update Bundler by running `gem install bundler`? Could not find gem 'bundler (< 2.0, >= 1.3.0)', which is required by gem 'rails (= 5.0.1)', in any of the sources. from C:/archivesspace/plugins/aspace-import-excel/gems/gems/bundler-2.0.1/lib/bundler/resolver.rb:56:in `start' from C:/archivesspace/plugins/aspace-import-excel/gems/gems/bundler-2.0.1/lib/bundler/resolver.rb:22:in `resolve' from C:/archivesspace/plugins/aspace-import-excel/gems/gems/bundler-2.0.1/lib/bundler/definition.rb:258:in `resolve' from C:/archivesspace/plugins/aspace-import-excel/gems/gems/bundler-2.0.1/lib/bundler/definition.rb:170:in `specs' from C:/archivesspace/plugins/aspace-import-excel/gems/gems/bundler-2.0.1/lib/bundler/definition.rb:237:in `specs_for' from C:/archivesspace/plugins/aspace-import-excel/gems/gems/bundler-2.0.1/lib/bundler/definition.rb:226:in `requested_specs' from C:/archivesspace/plugins/aspace-import-excel/gems/gems/bundler-2.0.1/lib/bundler/runtime.rb:108:in `block in requested_specs' from C:/archivesspace/plugins/aspace-import-excel/gems/gems/bundler-2.0.1/lib/bundler/runtime.rb:20:in `setup' from C:/archivesspace/plugins/aspace-import-excel/gems/gems/bundler-2.0.1/lib/bundler.rb:107:in `setup' from C:/archivesspace/plugins/aspace-import-excel/gems/gems/bundler-2.0.1/lib/bundler/setup.rb:20:in `<main>' from org/jruby/RubyKernel.java:961:in `require' from uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:1:in `(root)' from uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:133:in `<main>' from org/jruby/RubyKernel.java:961:in `require' from C:/archivesspace/data/tmp/jetty-0.0.0.0-8080-frontend.war-_-any-/webapp/WEB-INF/config/boot.rb:10:in `(root)' from uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:1:in `(root)' from uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:55:in `require' from uri:classloader:/jruby/rack/rails/environment3.rb:23:in `<main>' from C:\archivesspace\launcher\launcher.rb:92:in `start_server' from C:\archivesspace\launcher\launcher.rb:168:in `main' from C:\archivesspace\launcher\launcher.rb:261:in `<main>'
You're seeing this error because you use JRuby::Rack::ErrorApp::ShowStatus.


I still am at a loss and looking forward to somehow resolve this issue. I have used Bobbi’s import plugin in the past with earlier versions, and saved much time and effort in my workflow … thanks for this of course, and I would be appreciative of any other possible solutions to get my ASpace up and running with this feature included.

Best regards,
Greg

Gregory D. Farr
Archivist

[signature_458563074]Episcopal Church in Connecticut
   The Commons
   290 Pratt Street | Box 52 | Meriden CT 06450
   p 203-639-3501 x135
   f 203-235-1008
gfarr@episcopalct.org
www.epicscopalct.org<http://www.epicscopalct.org>

CONFIDENTIALITY NOTICE:
The contents of this email message and any attachments are intended solely for the addressee(s) and may contain confidential and/or privileged information and may be legally protected from disclosure. If you are not the intended recipient of this message or their agent, or if this message has been addressed to you in error, please immediately alert the sender by reply email and then delete this message and any attachments. If you are not the intended recipient, you are hereby notified that any use, dissemination, copying, or storage of this message or its attachments is strictly prohibited.

**Comment by bobbi-SMR** [comment]
Hi, Greg,

Would you attach the Gemfile.lock file that should be directly under the \aspace-import-excel\ folder?
Thanks

**Comment by gdfarr** [comment]
Hi Bobbi,

I’m attaching both gemfiles … hoping at least one is correct 😊

Gregory D. Farr
Archivist

[signature_1874363614]Episcopal Church in Connecticut
   The Commons
   290 Pratt Street | Box 52 | Meriden CT 06450
   p 203-639-3501 x135
   f 203-235-1008
gfarr@episcopalct.org
www.epicscopalct.org<http://www.epicscopalct.org>

CONFIDENTIALITY NOTICE:
The contents of this email message and any attachments are intended solely for the addressee(s) and may contain confidential and/or privileged information and may be legally protected from disclosure. If you are not the intended recipient of this message or their agent, or if this message has been addressed to you in error, please immediately alert the sender by reply email and then delete this message and any attachments. If you are not the intended recipient, you are hereby notified that any use, dissemination, copying, or storage of this message or its attachments is strictly prohibited.

**Comment by bobbi-SMR** [comment]
Hi, Greg,

I looked at the Gemfile.lock file you sent me, and it doesn't look like the one we expected to see.  Please confirm that you performed (or re-perform) the following steps:

1.  Delete the plugins\aspace-import-excel folder
2.  Download the zip file at https://github.com/harvard-library/aspace-import-excel/archive/issue_46.zip
3.  Extract the zip file into \plugins, then rename the folder from aspace-import-excel-issue_46 to aspace-import-excel
4.  run the initialize-plugins script.

After that, we would expect the Gemfile.lock to look exactly like https://github.com/harvard-library/aspace-import-excel/blob/issue_46/Gemfile.lock , which isn't what you sent me.

I'm going to be off-line for a few hours, but my colleague @pobocks will be monitoring the situation.

Sorry this is such a hassle.

**Comment by gdfarr** [comment]
Hi Bobbi,

Here is the gemfile.lock output that I have after re-install along the outlined steps below.

Thanks!
Greg

Gregory D. Farr
Archivist

[signature_1819214048]Episcopal Church in Connecticut
   The Commons
   290 Pratt Street | Box 52 | Meriden CT 06450
   p 203-639-3501 x135
   f 203-235-1008
gfarr@episcopalct.org
www.epicscopalct.org<http://www.epicscopalct.org>

CONFIDENTIALITY NOTICE:
The contents of this email message and any attachments are intended solely for the addressee(s) and may contain confidential and/or privileged information and may be legally protected from disclosure. If you are not the intended recipient of this message or their agent, or if this message has been addressed to you in error, please immediately alert the sender by reply email and then delete this message and any attachments. If you are not the intended recipient, you are hereby notified that any use, dissemination, copying, or storage of this message or its attachments is strictly prohibited.

**Comment by bobbi-SMR** [comment]
Hi, Greg,

You'll have to email it to my address, since it seems attachments don't come through when you send email through GitHub.  Or go to this issue in GitHub, and attach or copy/paste there.

**Comment by pobocks** [comment]
Hey, could you try deleting the plugin entirely and using the one from the zip-file, one more time, verifying:

- that the directory includes a Gemfile.lock BEFORE you run the install
- that that file looks like:

```
GEM
  remote: http://rubygems.org/
  specs:
    mini_portile2 (2.4.0)
    nokogiri (1.10.0)
      mini_portile2 (~> 2.4.0)
    rubyXL (3.3.29)
      nokogiri (>= 1.4.4)
      rubyzip (>= 1.1.6)
    rubyzip (1.2.2)

PLATFORMS
  ruby

DEPENDENCIES
  rubyXL (= 3.3.29)

BUNDLED WITH
   1.17.1
```

**Comment by gdfarr** [comment]
Yes, that is what it looks like BEFORE the install. Do you want me to try to run install again?


Gregory D. Farr
Archivist

[signature_1749231538]Episcopal Church in Connecticut
   The Commons
   290 Pratt Street | Box 52 | Meriden CT 06450
   p 203-639-3501 x135
   f 203-235-1008
gfarr@episcopalct.org
www.epicscopalct.org<http://www.epicscopalct.org>

CONFIDENTIALITY NOTICE:
The contents of this email message and any attachments are intended solely for the addressee(s) and may contain confidential and/or privileged information and may be legally protected from disclosure. If you are not the intended recipient of this message or their agent, or if this message has been addressed to you in error, please immediately alert the sender by reply email and then delete this message and any attachments. If you are not the intended recipient, you are hereby notified that any use, dissemination, copying, or storage of this message or its attachments is strictly prohibited.

**Comment by pobocks** [comment]
Yes please.

**Comment by gdfarr** [comment]
This is what the gemfile lock file looks after running script initial-plugin.bat aspace-import-excel :

GEM
  remote: http://rubygems.org/
  specs:
    mini_portile2 (2.4.0)
    nokogiri (1.10.0)
      mini_portile2 (~> 2.4.0)
    nokogiri (1.10.0-java)
    rubyXL (3.3.29)
      nokogiri (>= 1.4.4)
      rubyzip (>= 1.1.6)
    rubyzip (1.2.2)

PLATFORMS
  java
  ruby

DEPENDENCIES
  rubyXL (= 3.3.29)

BUNDLED WITH
   2.0.1

After re-starting service upon changing config file to recognize plugin directory ‘aspace-import-excel’ the following message appears n browser for staff-interface side (PUI remains unchanged and is operating as it should):

Internal Server Error (500)
Request Method:

GET

Request URL:

http://ecct6:8080/

Bundler could not find compatible versions for gem "bundler": In Gemfile: rails (= 5.0.1) java was resolved to 5.0.1, which depends on bundler (< 2.0, >= 1.3.0) java Current Bundler version: bundler (2.0.1) This Gemfile requires a different version of Bundler. Perhaps you need to update Bundler by running `gem install bundler`? Could not find gem 'bundler (< 2.0, >= 1.3.0)', which is required by gem 'rails (= 5.0.1)', in any of the sources. from C:/archivesspace/plugins/aspace-import-excel/gems/gems/bundler-2.0.1/lib/bundler/resolver.rb:56:in `start' from C:/archivesspace/plugins/aspace-import-excel/gems/gems/bundler-2.0.1/lib/bundler/resolver.rb:22:in `resolve' from C:/archivesspace/plugins/aspace-import-excel/gems/gems/bundler-2.0.1/lib/bundler/definition.rb:258:in `resolve' from C:/archivesspace/plugins/aspace-import-excel/gems/gems/bundler-2.0.1/lib/bundler/definition.rb:170:in `specs' from C:/archivesspace/plugins/aspace-import-excel/gems/gems/bundler-2.0.1/lib/bundler/definition.rb:237:in `specs_for' from C:/archivesspace/plugins/aspace-import-excel/gems/gems/bundler-2.0.1/lib/bundler/definition.rb:226:in `requested_specs' from C:/archivesspace/plugins/aspace-import-excel/gems/gems/bundler-2.0.1/lib/bundler/runtime.rb:108:in `block in requested_specs' from C:/archivesspace/plugins/aspace-import-excel/gems/gems/bundler-2.0.1/lib/bundler/runtime.rb:20:in `setup' from C:/archivesspace/plugins/aspace-import-excel/gems/gems/bundler-2.0.1/lib/bundler.rb:107:in `setup' from C:/archivesspace/plugins/aspace-import-excel/gems/gems/bundler-2.0.1/lib/bundler/setup.rb:20:in `<main>' from org/jruby/RubyKernel.java:961:in `require' from uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:1:in `(root)' from uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:133:in `<main>' from org/jruby/RubyKernel.java:961:in `require' from C:/archivesspace/data/tmp/jetty-0.0.0.0-8080-frontend.war-_-any-/webapp/WEB-INF/config/boot.rb:10:in `(root)' from uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:1:in `(root)' from uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:55:in `require' from uri:classloader:/jruby/rack/rails/environment3.rb:23:in `<main>' from C:\archivesspace\launcher\launcher.rb:92:in `start_server' from C:\archivesspace\launcher\launcher.rb:168:in `main' from C:\archivesspace\launcher\launcher.rb:261:in `<main>'
You're seeing this error because you use JRuby::Rack::ErrorApp::ShowStatus.


Gregory D. Farr
Archivist

[signature_2122779005]Episcopal Church in Connecticut
   The Commons
   290 Pratt Street | Box 52 | Meriden CT 06450
   p 203-639-3501 x135
   f 203-235-1008
gfarr@episcopalct.org
www.epicscopalct.org<http://www.epicscopalct.org>

CONFIDENTIALITY NOTICE:
The contents of this email message and any attachments are intended solely for the addressee(s) and may contain confidential and/or privileged information and may be legally protected from disclosure. If you are not the intended recipient of this message or their agent, or if this message has been addressed to you in error, please immediately alert the sender by reply email and then delete this message and any attachments. If you are not the intended recipient, you are hereby notified that any use, dissemination, copying, or storage of this message or its attachments is strictly prohibited.

**Comment by bobbi-SMR** [comment]
Hi,

If you're on the archivesspace-users mailing list, you will already have seen that we don't know (yet) how to fix this problem; I'm sorry for the delay. It seems that the new Bundler release isn't Playing Well With Others, despite their original assurances. We're in touch with lyrasis about this, as a fix may need to "come from the top" (we suspect that initializing any plugin with a Gemfile will have this problem! :-(

**Comment by pobocks** [comment]
See also #47

**Comment by bobbi-SMR** [comment]
Hi, Greg,

I *may* have a solution to installing aspace-import-excel on Windows.  Would you be willing to spend a little experimenting with me?

If so, I have a script to confirm what your bundler version (in the core ArchivesSpace), then, if it looks good, a modified initialize-plugin script, which *I*, at least, have been able to use to successfully install it on my local Windows repo.

**Comment by gdfarr** [comment]
Yes, sure … I will try 😊

Gregory D. Farr
Archivist

[signature_408208633]Episcopal Church in Connecticut
   The Commons
   290 Pratt Street | Box 52 | Meriden CT 06450
   p 203-639-3501 x135
   f 203-235-1008
gfarr@episcopalct.org
www.epicscopalct.org<http://www.epicscopalct.org>

CONFIDENTIALITY NOTICE:
The contents of this email message and any attachments are intended solely for the addressee(s) and may contain confidential and/or privileged information and may be legally protected from disclosure. If you are not the intended recipient of this message or their agent, or if this message has been addressed to you in error, please immediately alert the sender by reply email and then delete this message and any attachments. If you are not the intended recipient, you are hereby notified that any use, dissemination, copying, or storage of this message or its attachments is strictly prohibited.

**Comment by bobbi-SMR** [comment]
I'm emailing the two scripts to you.

**Comment by bobbi-SMR** [comment]
Ok,  
We solved Greg's problem by a combination of the issue_46 branch, which will be incorporated into a new release, and by using a modified *\scripts\initialize-plugin.bat* that replicates the bundler version detection that is found in _scripts/initialize-plugin.**sh**_

I had Greg delete the **aspace-import-excel**  folder, and then unzip the _issue_46_ branch zip file.

After that, he ran the following script, which he had placed in the **scripts** folder. @lorawoodford, I'm not sure how to get this into the distribution; I'm thinking that I should add an **etc/** directory to my repo with extensive directions on its use until it gets absorbed into the core base.

Here's **modified_initialize-plugin.bat**  
```dosbatch
@echo off 

SETLOCAL ENABLEDELAYEDEXPANSION

cd /d %~dp0..\plugins\%1

for /d %%a in (..\..\gems\gems\bundler-*) do set bnm=%%a
for /f "tokens=1* delims=-" %%a in ("%bnm%") do set vers=%%b
echo %vers%

set JRUBY=
FOR /D %%c IN (..\..\gems\gems\jruby-*) DO (
  set JRUBY=!JRUBY!;%%c\lib\*
)

set GEM_HOME=gems
java %JAVA_OPTS% -cp "..\..\lib\*!JRUBY!" org.jruby.Main -S gem install bundler -v "%vers%"
java %JAVA_OPTS% -cp "..\..\lib\*!JRUBY!" org.jruby.Main -S ..\..\gems\bin\bundle install --gemfile=Gemfile
```

**Comment by bobbi-SMR** [comment]
Ok,  
We solved Greg's problem by a combination of the issue_46 branch, which will be incorporated into a new release, and by using a modified *\scripts\initialize-plugin.bat* that replicates the bundler version detection that is found in _scripts/initialize-plugin.**sh**_

I had Greg delete the **aspace-import-excel**  folder, and then unzip the _issue_46_ branch zip file.

After that, he ran the following script, which he had placed in the **scripts** folder. @lorawoodford, I'm not sure how to get this into the distribution; I'm thinking that I should add an **etc/** directory to my repo with extensive directions on its use until it gets absorbed into the core base.

Here's **modified_initialize-plugin.bat**  
```dosbatch
@echo off 

SETLOCAL ENABLEDELAYEDEXPANSION

cd /d %~dp0..\plugins\%1

for /d %%a in (..\..\gems\gems\bundler-*) do set bnm=%%a
for /f "tokens=1* delims=-" %%a in ("%bnm%") do set vers=%%b
echo %vers%

set JRUBY=
FOR /D %%c IN (..\..\gems\gems\jruby-*) DO (
  set JRUBY=!JRUBY!;%%c\lib\*
)

set GEM_HOME=gems
java %JAVA_OPTS% -cp "..\..\lib\*!JRUBY!" org.jruby.Main -S gem install bundler -v "%vers%"
java %JAVA_OPTS% -cp "..\..\lib\*!JRUBY!" org.jruby.Main -S ..\..\gems\bin\bundle install --gemfile=Gemfile
```

**Comment by bobbi-SMR** [comment]
Addressed in [Release v2.1.18](https://github.com/harvard-library/aspace-import-excel/releases/tag/v2.1.18).

In addition, a [Pull Request](https://github.com/archivesspace/archivesspace/pull/1454) was made against core ArchivesSpace code to fix the problem at the source.
